spec/support/authenticated_session_helper.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
```

Comment:
@aub that's generally speaking fine with me. I built it with expect vs allow because I thought in 99% of specs it would only be called one time. This particular example is a rare case in which it is called 20 times, but I see your point. And instead, our generic test of authentication could have the necessary expectations.

I presume you're going to have problems with allow_any_instance_of still though. It's because of how Rspec treats this within an example block. It works under the assumption that you are stubbing the method for a single instance of the class, not every instance of the class.